monitoring/oplog-populator/dashboard.py

```diff
@@ -0,0 +1,335 @@
+    orientation='horizontal',
+    graphMode='none',
+    targets=[Target(
+        expr='sum by(connector) (' + Metrics.PIPELINE_SIZE.raw() + ')',
+        legendFormat='{{connector}}',
+    )],
+    thresholds=[
+        Threshold('green', 0, 0.),
+    ])
+
+bucket_additions = TimeSeries(
+    title='Bucket Addition Rate',
+    description='Rate at which buckets are getting added to the connectors',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    fillOpacity=10,
+    spanNulls=True,
+    unit=UNITS.OPS_PER_SEC,
+    targets=[Target(
+        expr='sum by(connector) (rate(' + Metrics.CONNECTOR_CONFIG(opType="add") + '))',
+        legendFormat='{{connector}}',
+    )])
+
+bucket_removals = TimeSeries(
+    title='Bucket Removal Rate',
+    description='Rate at which buckets are getting removed from the connectors',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    fillOpacity=10,
+    spanNulls=True,
+    unit=UNITS.OPS_PER_SEC,
+    targets=[Target(
+        expr='sum by(connector) (rate(' + Metrics.CONNECTOR_CONFIG(opType="delete") + '))',
+        legendFormat='{{connector}}',
+    )])
+
+ack_lag = TimeSeries(
+    title='Acknowledgement Lag By Operation Type',
+    description='Delta between a config change in mongo and it getting processed by the oplogPopulator in seconds',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    fillOpacity=10,
+    spanNulls=True,
+    unit=UNITS.SECONDS,
+    targets=[Target(
+        expr='sum by(opType) (rate(' + Metrics.ACK_LAG.sum() + '))',
+        legendFormat='{{opType}}',
+    )])
+
+reconfig_lag = TimeSeries(
+    title='Reconfig Lag By Connector',
+    description='Time it takes to reconfigure a connector in seconds',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    fillOpacity=10,
+    spanNulls=True,
+    unit=UNITS.SECONDS,
+    targets=[Target(
+        expr='sum by(connector) (rate(' + Metrics.RECONFIG_LAG.sum() + '))',
+        legendFormat='{{connector}}',
+    )])
+
+bucket_per_connector = TimeSeries(
+    title='Buckets Per Connector',
+    description='Evolution of buckets per connector',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    fillOpacity=10,
+    spanNulls=True,
+    targets=[Target(
+        expr='sum by(connector) (' + Metrics.BUCKETS.raw() + ')',
+        legendFormat='{{connector}}',
+    )])
+
+request_body_size_per_connector = TimeSeries(
+    title='Request Body Size Per Connector',
+    description='Size of kafka connect request in bytes',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    fillOpacity=10,
+    spanNulls=True,
+    unit=UNITS.BYTES_SEC,
+    targets=[Target(
+        expr='sum by(connector) (rate(' + Metrics.REQUEST_SIZE() + '))',
```

Comment:
that metric should be a GAUGE (since the request size _will_ go down, when we stop monitoring some buckets) : so we must not use `rate()`

on top of that, I guess just displaying the value is fine in this case : we basically just want to see the value, which should generally not evolve much, and possibly compare it with the mongodb limit...

Reply:
This metric contains the accumulation of all request body sizes hence making it a counter metric, we can also use it to get the overall bandwidth used by the oplog populator.